core/server/src/main/java/alluxio/master/file/options/CreatePathOptions.java

```diff
@@ -39,6 +40,7 @@ protected CreatePathOptions() {
     mPersisted = false;
     mRecursive = false;
     mMetadataLoad = false;
+    mDefaultMode = true;
```

Comment:
on a related note, I think `setPermission` should also set mDefaultMode to `false` at which point I think `mDefaultMode` could be made private and we can remove its setters and getters